openapi-integration-tests/src/test/java/io/ballerina/openapi/OpenAPITest.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.ballerina.openapi;
+
+import io.ballerina.openapi.cmd.TestUtil;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterTest;
+import org.testng.annotations.BeforeClass;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Comparator;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * This class to handle temp dirs and outputs.
+ */
+public abstract class OpenAPITest {
```

Comment:
Why do we need to make this abstract?

Reply:
Because this is used to provide common methods implementation to all the subclasses and future test subclasses.

Reply:
But we don't have any abstract methods defined within the class, do we?

Reply:
Yeah, currently we haven't any abstract method since those common methods have a default implementation.